plugins/janus_audiobridge.c

```diff
@@ -1503,17 +1535,35 @@ void janus_audiobridge_incoming_rtp(janus_plugin_session *handle, int video, cha
 		}
 		/* Enqueue the decoded frame */
 		janus_mutex_lock(&participant->qmutex);
-		g_queue_push_tail(participant->inbuf, pkt);
-		/* Keep the queue at max of 150 packets, though */
-		if(g_queue_get_length(participant->inbuf) > 150) {
-			/* Remove the oldest packet */
-			janus_audiobridge_rtp_relay_packet *pkt = g_queue_pop_head(participant->inbuf);
-			if(pkt != NULL) {
-				if(pkt->data)
-					g_free(pkt->data);
-				pkt->data = NULL;
-				g_free(pkt);
-				pkt = NULL;
+		/* Insert packets sorting by sequence number */
+		participant->inbuf = g_list_insert_sorted(participant->inbuf, pkt, &janus_audiobridge_rtp_sort);
+		if(participant->prebuffering) {
+			/* Still pre-buffering: do we have enough packets now? */
+			if(g_list_length(participant->inbuf) == DEFAULT_PREBUFFERING) {
+				participant->prebuffering = FALSE;
+				JANUS_LOG(LOG_VERB, "Prebuffering done! Finally adding the user to the mix\n");
+			} else {
+				JANUS_LOG(LOG_VERB, "Still prebuffering (got %d packets), not adding the user to the mix yet\n", g_list_length(participant->inbuf));
+			}
+		} else {
+			/* Make sure we're not queueing too many packets: if so, get rid of the older ones */
+			if(g_list_length(participant->inbuf) >= DEFAULT_PREBUFFERING*2) {
```

Comment:
@lminiero is the `DEFAULT_PREBUFFERING*2` here very specific or just a value that should work? 

I've been hearing choppyness in calls where I think that packets are getting dropped off the queue too soon. Could it be that this queue should be larger when the quality is increased, that would increase the number of flowing packets right?